src/utils/leaflet.js

```diff
@@ -0,0 +1,30 @@
+export const calculateMapBounds = points => {
+  let latitudes;
+  let longitudes;
+
+  const padding = 0.006;
+
+  if (!points.length) {
+    return null;
+  }
+
+  if (points.length === 1) {
+    latitudes = [points[0].latitude - padding, points[0].latitude + padding];
+    longitudes = [points[0].longitude - padding, points[0].longitude + padding];
+  }
+
+  if (points.length >= 2) {
+    latitudes = points.map(point => point.latitude);
+    longitudes = points.map(point => point.longitude);
+  }
+
+  const minLat = Math.min(...latitudes);
+  const minLng = Math.min(...longitudes);
+  const maxLat = Math.max(...latitudes);
+  const maxLng = Math.max(...longitudes);
+
+  return [
+    [minLat, minLng],
+    [maxLat, maxLng],
+  ];
```

Comment:
You could also make this less verbose (16 lines vs 28) and more self-explanatory:

```javascript
  if (!points.length) return null;

  // Explain why you are adding extra points in case there's only one
  const padding = 0.006;
  const pointsNormalized = points.length >= 2 ? points : [
    { latitude: points[0].latitude - padding, longitude: points[0].longitude - padding },
    { latitude: points[0].latitude + padding, longitude: points[0].longitude + padding }, 
  ]

  const latitudes = points.map(point => point.latitude);
  const longitudes = points.map(point => point.longitude);

  return [
    [Math.min(...latitudes), Math.min(...longitudes)],
    [Math.max(...latitudes), Math.max(...longitudes)],
  ];
```

Reply:
Thanks! Changed the code and also add a comment explaining the usage of padding